cupy/core/core.pyx

```diff
@@ -1242,12 +1242,14 @@ cdef class ndarray:
             return NotImplemented
 
     def __array_function__(self, func, types, args, kwargs):
-        if not hasattr(cupy, func.__name__):
+        module_list = '.'.join(func.__module__.split('.')[1:])
+        cupy_module = eval('cupy.' + module_list if module_list else 'cupy')
```

Comment:
The `eval` function is often avoided, sometimes due to security (though I'm not sure that's relevant here) but more often just to be explicit.

I wonder if the following would work instead
```python
import cupy as module
for submodule in func.__module__.split('.')[1:]:
    module = getattr(module, submodule)
```

We should probably also catch any `ImportError` or `AttributeError`s and return `NotImplemented` if the module doesn't exist in CuPy.

(this is only one way to solve this problem.  I wouldn't be surprised if the cupy maintainers have other suggestions.)

Reply:
Thank you for PR & suggestion!
@mrocklin's suggestion looks good to me.
 Yes, we should catch `AttributeError`, but I guess not `ImportError` (`cupy` is already imported in the global scope).